osu.Game/Overlays/Chat/DrawableChannel.cs

```diff
@@ -243,5 +235,32 @@ public DaySeparator(DateTimeOffset time)
                 };
             }
         }
+
+        /// <summary>
+        /// An <see cref="OsuScrollContainer"/> with functionality to automatically scrolls whenever the maximum scrollable distance increases.
+        /// </summary>
+        private class ChannelScrollContainer : OsuScrollContainer
+        {
+            private const float auto_scroll_leniency = 10f;
```

Comment:
xmldoc explaining what this value is (and what implications this has, why 10f is going to work 100% of the time).

Reply:
well, I took it directly from the previous auto-scrolling logic, perhaps the constant name isn't quite clear, it's a tolerance value for whether auto-scrolling should be performed, so user doesn't need to be 100% scrolled to end.

does this do it:
```csharp
/// <summary>
/// The least scroll value from end to consider auto-scrolling on.
/// </summary>
```

Reply:
```suggestion
            /// <summary>
            /// The chat will be automatically scrolled to end if and only if
            /// the distance between the current scroll position and the end of the scroll
            /// is less than this value.
            /// </summary>
            private const float auto_scroll_leniency = 10f;
```

that said 10 (pixels?) doesn't sound like a lot. other places use `UserTrackingScrollContainer` for a similar use-case, but the chat would need a "Return to newest" button thing for it to be applicable here. maybe something to follow up on later.